tests/testAutopas/ParticleIteratorTest.cpp

```diff
@@ -201,4 +201,83 @@ TEST_F(ParticleIteratorTest, testRMMIterator_mutable) {
       ASSERT_EQ(force[2], i);
     }
   }
+}
+
+template <class Container, class Molecule>
+void testContainerIteratorBehavior(Container& container, Molecule& mol, Molecule& haloMol){
```

Comment:
So this function expects that the container has two particles, one in the halo and one not, right? Maybe document this.

Reply:
added documentation.